builtin/providers/aws/resource_aws_api_gateway_authorizer.go

```diff
@@ -200,7 +201,11 @@ func resourceAwsApiGatewayAuthorizerDelete(d *schema.ResourceData, meta interfac
 	log.Printf("[INFO] Deleting API Gateway Authorizer: %s", input)
 	_, err := conn.DeleteAuthorizer(&input)
 	if err != nil {
-		return fmt.Errorf("Deleting API Gateway Authorizer failed: %s", err)
+		// XXX: Figure out a way to delete the method that depends on the authorizer first
+		// otherwise the authorizer will be dangling until the API is deleted
+		if !strings.Contains(err.Error(), "ConflictException") {
```

Comment:
As long as the authorizer is managed in the same terraform scope and is referenced properly (i.e. **not hardcoded** authorizer IDs in methods) Terraform will schedule the deletion correctly by default - i.e. for complete destruction `1st method => 2nd authorizer`. Have a look at the dependency graph via `terraform graph`. 😉 

The only issue that may theoretically arise is eventual consistency of the AWS API (or the implementation of it) - i.e. the `authorizer_id` change may take time to propagate. We usually just retry the deletion in such cases with a reasonable timeout.

Did you experience this problem yourself @johnjelinek ?